lib/manager/batect/extract.ts

```diff
@@ -103,3 +136,39 @@ export function extractPackageFile(
     return null;
   }
 }
+
+export async function extractAllPackageFiles(
+  config: ExtractConfig,
+  packageFiles: string[]
+): Promise<PackageFile[] | null> {
+  const filesToExamine = [...packageFiles];
+  const filesAlreadyExamined = [];
+  const results: PackageFile[] = [];
+
+  while (filesToExamine.length > 0) {
+    const packageFile = filesToExamine.pop();
+    filesAlreadyExamined.push(packageFile);
+
+    const content = await readLocalFile(packageFile, 'utf8');
+    const result = extractPackageFile(content, packageFile);
+
+    if (result !== null) {
+      result.referencedConfigFiles.forEach((f) => {
+        if (
+          filesAlreadyExamined.indexOf(f) === -1 &&
+          filesToExamine.indexOf(f) === -1
+        ) {
+          filesToExamine.push(f);
+        }
+      });
+
+      results.push({
+        manager: 'batect',
+        packageFile,
```

Comment:
Overriding the packageFile value is not something we have done before, and needs careful evaluation in case it causes any undesirable effects

Reply:
Yeah, but if this works, we can use it in `gitlab-include` too.

Reply:
I've removed the override of `manager` in da6e077ad86c31976129fce362479db16c00a81a.

For `packageFile`, is there something else you'd suggest?

Reply:
@rarkins Other managers (maven / bundler / ... ) are using this too. Note, we are using `extractAllPackageFiles` here not `extractPackageFile`.

Looks like `PackageFile.manager` is never used
https://github.com/renovatebot/renovate/blob/303449a897f8bfcc3db9fbf1f30f79a1b0694552/lib%2Fmanager%2Fcommon.ts#L93

So i think the following can be safely removed too
https://github.com/renovatebot/renovate/blob/11371121306b8c57ee039af8e067356bbcd92d91/lib%2Fworkers%2Frepository%2Fextract%2Fmanager-files.ts#L63

Reply:
OK, I'm fine with this then. Makes sense with extractAllPackageFiles, I assume that is to perform de-duplication too?

Reply:
Yes, that's why the loop saves processed and todo file lists.